Common/RoguelikeChange/RoguelikeItemOverhaul.cs

```diff
@@ -91,148 +35,51 @@ class RoguelikeItemOverhaul : GlobalItem {
-					   "\nOver charged: Increases movement speed, weapon speed, damage by 10%";
-			}
-			if (type == ItemID.PearlwoodHelmet || type == ItemID.PearlwoodBreastplate || type == ItemID.PearlwoodGreaves) {
-				return "Increase movement speed by 35%" +
-						"\nAttacking an enemy summons 6 hallow Swords that deals 5 damage with 4 seconds cooldown" +
-						"\nIncrease damage by 15% during day" +
-						"\nIncrease defense By 12" +
-						"\nWhen in Hallow biome:" +
-						"\n Hallow Swords deal 15 damage";
-			}
-			if (type == ItemID.IronHelmet || type == ItemID.IronChainmail || type == ItemID.IronGreaves) {
-				return "Increase damage reduction 2.5%" +
-					   "\nIncrease defense effectivness by 10%" +
-					   "\nIncrease damage by 5%" +
-					   "\nDecrease movement speed 5%" +
-					   "\nWhile under 50% HP, gain 15 bonus defense, but -10% less attack speed";
-			}
-			if (type == ItemID.SilverHelmet || type == ItemID.SilverChainmail || type == ItemID.SilverGreaves) {
-				return "During the day :" +
-					   "\nGain 10 defense" +
-					   "\nDuring the night :" +
-					   "\nIncrease damage by 10%" +
-					   "\nAt full life, these effects are multiply by 2";
-			}
-			if (type == ItemID.TungstenHelmet || type == ItemID.TungstenChainmail || type == ItemID.TungstenGreaves) {
-				return "Increase defense by 15" +
-					   "\nWhen at full hp :" +
-					   "\nReduce your defense down to 0" +
-					   "\nIncrease speed by 30%" +
-					   "\nThe closer your enemy is, the more damage increases";
-			}
-			if (type == ItemID.GoldHelmet || type == ItemID.GoldChainmail || type == ItemID.GoldGreaves) {
-				return "Your attack have 15% to inflict Midas for 10 seconds" +
-					   "\nAttacking enemies with midas debuff will : " +
-					   "\nDeal additional damage based on their defense";
-			}
-			if (type == ItemID.PlatinumHelmet || type == ItemID.PlatinumChainmail || type == ItemID.PlatinumGreaves) {
-				return "Increase weapon uses speed by 35%" +
-					   "\nAttacking too much will lit on fire";
-			}
-			return "";
-		}
 		public override string IsArmorSet(Item head, Item body, Item legs) {
 			if (!ModContent.GetInstance<BossRushModConfig>().RoguelikeOverhaul) {
 				return "";
 			}
 			return new ArmorSet(head.type, body.type, legs.type).ToString();
 		}
+		//I really need to make this whole GetToolTip and UpdateArmorSet to be somehow it's own classes, maybe utilize ArmorSet class ?
+		private string GetToolTip(int type, int[] armorSet) {
+			ArmorSet set = new ArmorSet(armorSet[0], armorSet[1], armorSet[2]);
+			if (set.ContainAnyOfArmorPiece(type)) {
+				return set.AttemptToGetToolTip();
+			}
+			return "";
+		}
 		public override void UpdateArmorSet(Player player, string set) {
 			GlobalItemPlayer modplayer = player.GetModPlayer<GlobalItemPlayer>();
-			if (set == ArmorSet.ConvertIntoArmorSetFormat(ItemID.WoodHelmet, ItemID.WoodBreastplate, ItemID.WoodGreaves)) {
+			if (set == new WoodArmorSet().ToString()) {
```

Comment:
`ArmorSet.ConvertIntoArmorSetFormat(ItemID.WoodHelmet, ItemID.WoodBreastplate, ItemID.WoodGreaves)` is preferred over creating new object.

Reply:
alright, I will revert it